### PR TITLE
More tests for data driven trx-Files

### PR DESCRIPTION
**Pull request by gfoidl**
Follow up to #38 
Adds some trx-Files with data driven tests.

### CODE REVIEW

File: source/trx2junit/Internal/JUnitBuilder.cs [hunk 1]
```diff
@@ -90,7 +90,7 @@ private void AddTest(XElement xTestSuite, TestDefinition test)
                     _failures++;
                     xTestCase.Add(new XElement("failure",
                         unitTestResult.StackTrace,
-                        new XAttribute("message", unitTestResult.Message),
+                        new XAttribute("message", unitTestResult.Message ?? ""),
```
Comment (gfoidl) [inline source/trx2junit/Internal/JUnitBuilder.cs]: Add a comment that `Message` is allowed to be `null`.

**Review by gfoidl — COMMENTED** [review]
Nit

**Comment by gfoidl** [comment]
@cheng93 FYI. 

NUnit and xUnit look good, and correct numbers are reported.
MsTest reports data driven tests in a different scheme, I logged https://github.com/gfoidl/trx2junit/issues/43 for this, so that PR (and your PR) are ready to ship.

Can you please have a look (at this PR, not the issue -- this will be addressed sometime...)?

**Comment by cheng93** [comment]
The data driven test looks fine, however they weren't my scenarios. Notice how the trx files have a 1 to 1 relation.

In xUnit using `MemberData` with anonymous object should produce a 1 to many.

Looks like MsTest have InnerResults

**Comment by cheng93** [comment]
See https://github.com/gfoidl/trx2junit/pull/45 for my scenario for xunit

**Review by cheng93 — APPROVED** [review]

**Comment by gfoidl** [comment]
Some counts (# of tests, # of failures) aren't correct (MsTest and xUnit data driven).
https://circleci.com/gh/gfoidl/trx2junit/174

Maybe it's the test itself, or some counts aren't collected correctly. Will investigate this...

**Comment by gfoidl** [comment]
### MsTest

Has `InnerResults`. https://github.com/gfoidl/trx2junit/blob/2b2a89050ff008c4bb2f509b11a6cb9a09bd435f/tests/trx2junit.Tests/data/mstest-datadriven.trx#L17-L31

The question is, should these be counted separately or not.
VS / Test Results view shows 3/6 passed. That seems odd, as there are only 5 tests. Maybe VS is here buggy? 

### xUnit

VS shows 3/5 passed, which is correct.
Test execution shows 3 tests (expected 5), with 1 failure (expected 2).

**Comment by cheng93** [comment]
xUnit

Looks like the testdefinition classname generated by the TRX is wrong.

https://github.com/xunit/visualstudio.xunit/issues/115

MsTest

For data driven test we probably want to exclude the wrapper test?

**Comment by gfoidl** [comment]
> MsTest
>
> For data driven test we probably want to exclude the wrapper test?

Agree, so this fixes https://github.com/gfoidl/trx2junit/issues/43

**Comment by gfoidl** [comment]
Now it seems correct, and CI is happy now :wink:

@cheng93 thanks for your help 👍

**Pull request merged**